frontend/src/hooks/usePrice.ts

```diff
@@ -22,31 +23,58 @@ export function useEtherPrice(blockIdentifier: BlockNumber = BlockTag.latest) {
   return useMemo(() => {
     if (!pairReserves.data) return
 
-    const ethPrice = new Fraction(
+    const pairPrice = new Fraction(
       uint256.uint256ToBN(pairReserves.data.reserve1).toString(),
       uint256.uint256ToBN(pairReserves.data.reserve0).toString()
     )
 
     // token0 and token1 are switched on goerli and mainnet ...
     return (
       chainId === constants.StarknetChainId.SN_GOERLI
-        ? new Fraction(ethPrice.denominator, ethPrice.numerator)
-        : ethPrice
+        ? new Fraction(pairPrice.denominator, pairPrice.numerator)
```

Comment:
depending on the pair, you don't always have to reverse the result. This should be stored in a const object